distribution/tools/entitlement-runtime/src/main/java/org/elasticsearch/entitlement/runtime/policy/FileEntitlementBuilder.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.entitlement.runtime.policy;
+
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.elasticsearch.entitlement.runtime.policy.PolicyParserException.newPolicyParserException;
+
+public class FileEntitlementBuilder extends EntitlementBuilder {
+
+    public static ParseField FILE_PARSEFIELD = new ParseField("file");
+    public static ParseField PATH_PARSEFIELD = new ParseField("path");
+    public static ParseField ACTIONS_PARSEFIELD = new ParseField("actions");
+
+    protected String path;
+    protected final List<String> actions = new ArrayList<>();
+
+    protected FileEntitlementBuilder(String policyName, String scopeName, XContentParser policyParser) {
+        super(policyName, scopeName, policyParser);
+    }
+
+    protected void parseEntitlement() throws IOException {
```

Comment:
NGL, I didn't read the implementation here in detail. This seems like a fairly low-stakes situation; I assume you did smart things in here, and if we find out something isn't right, we can just fix it. An extensive test suite is probably better for demonstrating that this sort of method is right, as opposed to careful review.

Reply:
I understand :) . Tests should cover this kind of thing typically.